OSDOCS-6528: Created documentation for creating a shared VPC (#63793)
openshift/openshift-docs: modules/rosa-sharing-vpc-hosted-zones.adoc

// Module included in the following assemblies:
//
// * networking/rosa-shared-vpc-config.adoc
:_content-type: PROCEDURE
[id="rosa-sharing-vpc-hosted-zones_{context}"]
= Updating the shared VPC role and creating hosted zones

After the *cluster-creating AWS account* user provides the DNS domain and the IAM roles, *the VPC-owning AWS account* user must create a private hosted zone and update the trust policy on the IAM role that was created for sharing the VPC.

.Procedure

> . The *VPC-owning AWS account* user who owns the VPC must update the VPC sharing IAM role and add the _Installer_ and _Ingress Operator Cloud Credentials_ roles to the principal section of the trust policy.

Reviewer: To more directly address the user with an imperative statement, consider adding a Prerequisites section and then rewriting this step. For example:
```
.Prerequisites

You must have the VPC-owning AWS account user role that owns the VPC.

.Procedure

. Update the VPC sharing IAM role and add the `Installer` and `Ingress Operator Cloud Credentials` roles to the principal section of the trust policy:
```

+
[source,terminal]
----
{
  "Version": "2012-10-17",
  "Statement": [
    {
	  "Sid": "Statement1",
	  "Effect": "Allow",
	  "Principal": {
	  	"AWS": [
          "arn:aws:iam::<Cluster-Creator's-AWS-Account-ID>:role/<prefix>-ingress-operator-cloud-credentials",
          "arn:aws:iam::<Cluster-Creator's-AWS-Account-ID>:role/<prefix>-Installer-Role"
        ]            
	  },
	  "Action": "sts:AssumeRole"
	}
  ]
}
----

> . After updating the trust policy, the *VPC-owning AWS account* user creates a private hosted zone in the link:https://us-east-1.console.aws.amazon.com/route53/v2/[Route 53 section of the AWS console]. In the hosted zone configuration, the domain name will be `<cluster-name>.<dns_domain>`. The private hosted zone needs to be associated with the created VPC.

Reviewer: ```suggestion
. After you update the trust policy, create a private hosted zone in the link:https://us-east-1.console.aws.amazon.com/route53/v2/[Route 53 section of the AWS console]. In the hosted zone configuration, the domain name is `<cluster-name>.<dns_domain>`. The private hosted zone must be associated with the created VPC.
```

> . Once the hosted zone is created and associated with the VPC, provide the following to the *cluster-creating AWS account* user:

Reviewer: ```suggestion
. After the hosted zone is created and associated with the VPC, provide the following to the *cluster-creating AWS account* user:
```
